### PR TITLE
Adding support for notifications

### PR DESCRIPTION
**Pull request by brimstone**
This PR will add support for notifications. Task list is as follows:
- [X] Actually pass a message upstream
- [x] Resolve getting device type from the stream or getting it from the user
- [x] Manage an internal timer for knowing when a timer completes
- [x] Pass alarm messages upstream
- [x] Update this item if something internal needs to be done to manage when alarms complete
- [x] Update README with example

### CODE REVIEW

**Comment by brimstone** [comment]
This is ready to be tested by someone else and merged.

File: index.js [hunk 1]
```diff
@@ -182,6 +197,17 @@ Reverb.prototype.onGatewayCommand = function(command) {
             id: body.key.entryId
           , user: body.key.registeredUserId
         });
+    } else if (command.command == 'PUSH_NOTIFICATION_CHANGE') {
+        var body = JSON.parse(command.payload);
+        if (body.notificationId == "t1") {
```
Comment (dhleong) [inline index.js:202]: Would it make more sense to switch on the first character here instead of testing the id exactly? In case there's ever a `t2`, etc.

**Comment by dhleong** [comment]
Cool stuff! Not sure when I'll be able to test it, but looks pretty solid at first glance. 

Looks like we might want to abstract out some of that Fetcher stuff soon if we keep up like this

File: index.js (hunk 1 above)
Comment (brimstone) [inline index.js:202]: Maybe. There's more assumptions of a single timer elsewhere. If they do have multiple timers later, a better way might be to have an onNotification, where the notification is resolved and the Type attribute is then used to figure out if an onTimer or onAlarm needs to happen next. I feel like this is shippable currently.

**Comment by dhleong** [comment]
LGTM! Thanks :)

**Pull request merged**